packages/angular-cli/models/webpack-build-utils.ts

```diff
@@ -109,3 +109,18 @@ export function extraEntryParser(
       return extraEntry;
     });
 }
+
+const hashFormats: { [option: string]: any } = {
+  'none':     { chunk: '',             extract: '',                  file: '[name]'    },
+  'media':    { chunk: '',             extract: '',                  file: '[hash:20]' },
```

Comment:
20 is the webpack default and the existing tests rely on it so I went with it for consistency.  Personally I'd prefer a small decrease in human readability in exchange for a lower chance of collision. Especially considering they're not intended to be overly human readable. (That's why we have dev builds.)
How about something in the middle: 12?
You're example is actually `[hash:32]` (or just` [hash]`) when using the file-loader.  Oddly, `[hash] ` means different things in different places in webpack and the defaults don't seem to carry over either.
I agree on the name inclusion as it would be consistent with the others but was hesitant to change the existing paradigm without a consensus.

@filipesilva thoughts?